packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1460,6 +1608,50 @@ export class Bucket extends BucketBase {
 
     this.accessControl = BucketAccessControl.LOG_DELIVERY_WRITE;
   }
+
+  private parseInventoryConfiguration(): CfnBucket.InventoryConfigurationProperty[] | undefined {
+    if (!this.inventories || this.inventories.length === 0) {
+      return undefined;
+    }
+
+    return this.inventories.map((inventory, index) => {
+      const format = inventory.format ?? InventoryFormat.CSV;
+      const frequency = inventory.frequency ?? InventoryFrequency.WEEKLY;
+      const id = inventory.inventoryId ?? `${this.node.id}Inventory${index}`;
+
+      if (inventory.destination.bucket instanceof Bucket) {
+        inventory.destination.bucket.addToResourcePolicy(new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ['s3:PutObject'],
+          resources: [
+            inventory.destination.bucket.bucketArn,
+            inventory.destination.bucket.arnForObjects('*'),
```

Comment:
Feels like this should consider the destination prefix - right?

```suggestion
            inventory.destination.bucket.arnForObjects(`${inventory.destination.prefix ?? ''}*`)
```